clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -1419,7 +1424,8 @@ void constrainConsVarGeq(ConstraintVariable *LHS, ConstraintVariable *RHS,
               n++;
             }
           // Unequal sizes means casting from (say) T** to T*; not safe.
-          } else {
+          // unless assigning to a generic type.
+          } else if (!(PCLHS->GetIsGeneric() || PCRHS->GetIsGeneric())) {
```

Comment:
Ah. Maybe say that this is the ultimate reason (that is, not constraining to be wild) that we have the IsGeneric flag?